### PR TITLE
(Sentinel) Sentinel peer check

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

Checks number of peers every 10 seconds, adds message if the number of peers have changed

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [x] Should publish Docker image

### CODE REVIEW

File: sentinel/pkg/checker/peers/app.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	return resp.Count, nil
+}
+
+func alarm(newCount int, oldCount int) {
+	if newCount > oldCount {
+		alert.SlackAlert(fmt.Sprintf("number of peers in orakl offchain cluster has increased from %d to %d", oldCount, newCount))
+	} else {
+		alert.SlackAlert(fmt.Sprintf("number of peers in orakl offchain cluster has decreased from %d to %d", oldCount, newCount))
+	}
+}
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply sentinel/pkg/checker/peers/app.go:87]: the value is not expected to have large fluctuation, should be okay without threshold

File: sentinel/pkg/checker/peers/app.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	initialCount, err := checkPeerCounts()
+	if err != nil {
+		return err
+	}
+	peerCount = initialCount
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply sentinel/pkg/checker/peers/app.go:41]: it is intended to use default interval if err occurs

File: sentinel/pkg/checker/peers/app.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	log.Info().Msg("Starting peer count checker")
+	checkTicker := time.NewTicker(peerCheckInterval)
+	defer checkTicker.Stop()
+
+	for range checkTicker.C {
+		newPeerCount, err := checkPeerCounts()
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to check peer count")
+			continue
+		}
```
Comment (Intizar-T) [inline sentinel/pkg/checker/peers/app.go:58]: how about counting how many times it fails checking peer count and if it passes certain threshold (maybe 5-10 times) then send a slack message?

**Review by Intizar-T — APPROVED** [review]
lgtm!

left a suggestion

**Pull request merged**